tests/experimental/datasources/test_utils.py

```diff
@@ -0,0 +1,377 @@
+                == "com.superconductive.rendered.string"
+            )
+        else:
+            assert r.rendered_content is None
+            assert r.expectation_config.rendered_content is None
+
+    # Rudimentary test for data doc generation
+    docs_dict = context.build_data_docs()
+    assert "local_site" in docs_dict, "build_data_docs returned dictionary has changed"
+    assert docs_dict["local_site"].startswith(
+        "file://"
+    ), "build_data_docs returns file path in unexpected form"
+    path = docs_dict["local_site"][7:]
+    with open(path) as f:
+        data_doc_index = f.read()
+
+    # Checking for ge-success-icon tests the result table was generated and it was populated with a successful run.
+    assert "ge-success-icon" in data_doc_index
+    assert "ge-failed-icon" not in data_doc_index
+
+
+def run_data_assistant_and_checkpoint(
+    datasource_test_data: tuple[
+        AbstractDataContext, Datasource, DataAsset, BatchRequest
+    ]
+):
+    context, ds, _, batch_request = datasource_test_data
+    data_assistant_result, checkpoint_result = _configure_and_run_data_assistant(
+        context, batch_request
+    )
+    batch_num = len(
+        data_assistant_result._batch_id_to_batch_identifier_display_name_map
+    )
+    assert batch_num == 1, f"Only expected 1 batch but found {batch_num}"
+
+    # We assert the data assistant successfully generated expectations.
+    # We don't care about the exact number since that may change as data assistants evolve.
+    expectation_num = len(data_assistant_result.expectation_configurations)
+    assert checkpoint_result.success, "Running expectation suite failed"
+    # Verify that the number of checkpoint validations is the number of expectations generated by the data assistant
+    assert (
+        len(
+            checkpoint_result["run_results"][
+                list(checkpoint_result["run_results"].keys())[0]
+            ]["validation_result"].results
+        )
+        == expectation_num
+    )
+
+
+def run_multibatch_data_assistant_and_checkpoint(
+    multibatch_datasource_test_data: tuple[
+        AbstractDataContext, Datasource, DataAsset, BatchRequest
+    ]
+):
+    context, ds, _, batch_request = multibatch_datasource_test_data
+    execution_engine = ds.execution_engine
+    data_assistant_result, checkpoint_result = _configure_and_run_data_assistant(
+        context, batch_request
+    )
+    # Assert multiple batches were processed
+    batch_num = len(
+        data_assistant_result._batch_id_to_batch_identifier_display_name_map
+    )
+    assert batch_num == 12, f"Expected exactly 12 batches but found {batch_num}"
+
+    expectation_num = len(data_assistant_result.expectation_configurations)
+    if isinstance(execution_engine, (PandasExecutionEngine, SparkDFExecutionEngine)):
+        assert expectation_num == 109
+
+    if isinstance(execution_engine, SqlAlchemyExecutionEngine):
+        assert expectation_num == 111
```

Comment:
These numbers should probably be constants and defined at the top of the file?

Reply:
@cdkini We have not done that in other tests that assert on specific counts (e.g., in the `DataAssistant` tests).  Thanks.